streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -110,20 +107,7 @@ public int hashCode() {
         ClientMetadata(final String endPoint) {
 
             // get the host info if possible
-            if (endPoint != null) {
-                final String host = getHost(endPoint);
-                final Integer port = getPort(endPoint);
-
-                if (host == null || port == null) {
-                    throw new ConfigException(
-                        String.format("Error parsing host address %s. Expected format host:port.", endPoint)
-                    );
-                }
-
-                hostInfo = new HostInfo(host, port);
-            } else {
-                hostInfo = null;
-            }
+            hostInfo = endPoint != null ? HostInfo.buildFromEndpoint(endPoint) : null;
```

Comment:
nit: I feel we could internalize the hostInfo creation, including the null check:
`hostInfo = HostInfo.buildFromEndpoint(endPoint);`

Reply:
Ack